src/gpu/generic/sycl/ref_matmul.cpp

```diff
@@ -52,16 +55,56 @@ void ref_matmul_t::pd_t::init_conf() {
     for (const auto &mdw : {src_d, weights_d, dst_d, bias_d}) {
         if (mdw.has_runtime_dims()) {
             any_runtime_params_ = true;
-            return;
+            return status::unimplemented;
         }
     }
-    init_rt_conf(conf_, src_d, weights_d, dst_d, bias_d);
+    init_rt_conf(conf_, data_md_t, dst_md_t, weights_md_t, src_d,
+            weights_d, dst_d, bias_d);
+    return status::success;
 }
 
 void ref_matmul_t::pd_t::init_rt_conf(sycl_matmul_conf_t &conf,
+        xpu::sycl::md_t_spec_const &data_md_t_,
+        xpu::sycl::md_t_spec_const &dst_md_t_,
+        xpu::sycl::md_t_spec_const &weights_md_t_,
         const memory_desc_wrapper src_d, const memory_desc_wrapper weights_d,
         const memory_desc_wrapper dst_d,
         const memory_desc_wrapper bias_d) const {
+
+    // Lambda because this function will not be used anywhere else
+    auto init_md_t_sc_from_md = [=](xpu::sycl::md_t_spec_const &md_t_sc,
+                                        const memory_desc_t *md) -> void {
+        constexpr int max_dims = 6;
+        using dim32_t = int32_t;
+
+        memory_desc_wrapper mdw(md);
+
+        assert(mdw.format_kind() == format_kind::blocked);
+        assert(mdw.ndims() <= max_dims);
```

Comment:
given that this function can fail, I would advocate to return a status instead of using asserts, and propagate by it in init_conf.

Reply:
Asserts changed to verbose checks, propagating status_t now as well.